### ISSUE TITLE
Thread hanging even after close

### ISSUE THREAD

**Issue by n-scope** [deepstreamIO/deepstream.io-client-java#98]
Hi all,
I am using java deepstreeam client 2.2.1, in a desktop environment (not Android).
When performing any rpc call, the app won't exit because some threads (used by deepstream) in the executor pool are still waiting for tasks. 
![image](https://user-images.githubusercontent.com/16793867/28241298-39d34adc-6992-11e7-87df-b44511080438.png)

Perhaps I am missing something, but I guess the ds.close() call should trigger a complete shutdown of the pool ?   

Here is the minimal code to reproduce this issue:
```
 public static void main(String[] pArgs) {
        try {
            DeepstreamClient ds = new DeepstreamClient("myhost");
            JsonObject auth = new JsonObject();
            auth.add("username", new JsonPrimitive("myuser"));
            auth.add("password", new JsonPrimitive("mypassword"));
            ds.login(auth);
            ds.rpc.make("anything", "data");
            ds.close();
            System.out.println("Main Thread END");
        } catch (Exception e) {
            e.printStackTrace();
        }
}
```

BTW, thanks again for this great project !

**Comment by yasserf** [comment]
Yeah that makes sense! Thank you for the reproducible code

**Comment by AlexBHarley** [comment]
Thanks for raising this, should be fixed in `v2.2.2`. Let us know if there are any problems.
